### PR TITLE
revert defaulter-gen

### PR DESCRIPTION
**Pull request by theishshah**
**Description of the change:**
Revert defaulter-gen changes made in #1050 

**Motivation for the change:**
Discussion on #1038

### CODE REVIEW

**Comment by hasbro17** [comment]
Can you reformat the commit message according to our [guidelines](https://github.com/operator-framework/operator-sdk/blob/master/CONTRIBUTING.MD#format-of-the-commit-message).

Also add a line in the [removed section](https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md#removed) of the CHANGELOG to point this out. And probably mention that defaulting for CRs should be done with mutating webhooks.

**Comment by joelanford** [comment]
Should we also remove `defaulter-gen` references from `internal/pkg/scaffold/gopkgtoml.go` and `sdk-cli-reference.md`?

**Comment by hasbro17** [comment]
@joelanford Yes good catch.
@theishshah Can you update the PR to remove the constraint for defaulter-gen from the scaffold:
https://github.com/operator-framework/operator-sdk/blob/master/internal/pkg/scaffold/gopkgtoml.go#L46
And update the example in the CLI reference as well:
https://github.com/operator-framework/operator-sdk/blob/master/doc/sdk-cli-reference.md#example-1

**Comment by hasbro17** [comment]
You need to update `gopkgtoml_test.go` to remove `"k8s.io/code-generator/cmd/defaulter-gen"` from the required imports as well.

File: version/version.go [hunk 1]
```diff
@@ -15,5 +15,5 @@
 package version
 
 var (
-	Version = "v0.6.0+git"
+	Version = "v0.7.0+git"
```
Comment (joelanford) [inline version/version.go]: @theishshah I'm confused why some of these 0.7.0 changes are showing up in this PR. I checked `master` to be sure and all of them are already there. Maybe some weirdness happened with merging or rebasing after 0.7.0 was released?

Comment (estroz) [reply version/version.go]: Yeah you might want to squash merge onto master.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 ### Removed
 
+- The SDK will no longer run `defaulter-gen` on running `operator-sdk generate k8s`. Defaulting for CRDs should be handled with mutating admission webhooks. ([#1288](https://github.com/operator-framework/operator-sdk/pull/1288))
```
Comment (joelanford) [inline CHANGELOG.md]: Nit: remove trailing whitespace.

```suggestion
- The SDK will no longer run `defaulter-gen` on running `operator-sdk generate k8s`. Defaulting for CRDs should be handled with mutating admission webhooks. ([#1288](https://github.com/operator-framework/operator-sdk/pull/1288))
```

**Pull request merged**